node/narwhal/src/primary.rs

```diff
@@ -1053,3 +1053,317 @@ impl<N: Network> Primary<N> {
         self.gateway.shut_down().await;
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use crate::MAX_EXPIRATION_TIME_IN_SECS;
+
+    use snarkos_node_narwhal_committee::MIN_STAKE;
+    use snarkos_node_narwhal_ledger_service::MockLedgerService;
+
+    use bytes::Bytes;
+    use rand::RngCore;
+
+    type CurrentNetwork = snarkvm::prelude::Testnet3;
+
+    // Returns a primary and a list of accounts in the configured committee.
+    async fn primary_without_handlers(
+        rng: &mut TestRng,
+    ) -> (Primary<CurrentNetwork>, Vec<(SocketAddr, Account<CurrentNetwork>)>) {
+        // Create a committee containing the primary's account.
+        let (accounts, committee) = {
```

Comment:
Is there a reason you can't use `snarkos_node_narwhal_committee::test_helpers::sample_committee(rng)`?

Reply:
Yes, I need the accounts as well (private keys) but I could maybe refactor this code to make use of the helper (or alternatively expand the helper) 🤔